lib/Reporters/ASTWhitelistSourceInfoProvider.cpp

```diff
@@ -0,0 +1,71 @@
+//
+// Created by Stanislav Pankevich on 2019-10-18.
+//
+
+#include "mull/Reporters/ASTWhitelistSourceInfoProvider.h"
+
+#include "mull/JunkDetection/CXX/ASTStorage.h"
+#include "mull/Logger.h"
+#include "mull/MutationPoint.h"
+
+#include <clang/AST/Expr.h>
+#include <clang/Lex/Lexer.h>
+
+#include <iostream>
+
+using namespace mull;
+
+ASTWhitelistSourceInfoProvider::ASTWhitelistSourceInfoProvider(
+    ASTStorage &astStorage, ASTMutations &mutations)
+    : astStorage(astStorage), mutations(mutations) {}
+
+MutationPointSourceInfo
+ASTWhitelistSourceInfoProvider::getSourceInfo(MutationPoint *mutationPoint) {
```

Comment:
This source info provider is implemented separately from the one used by Element because of the following:

- it uses `ASTMutations` to get the AST nodes and their locations collected by AST Search instead of using the information that is collected by Junk Detection.